styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java

```diff
@@ -58,6 +59,24 @@ private boolean isUserError(Throwable e) {
 
   @Override
   public void transitionInto(RunState state, EventRouter eventRouter) {
+    switch (state.state()) {
```

Comment:
There is a duplication of code in this part between both Execution handlers, and also in this class `createRunSpec` still does the checks that is in this block of code